vars/sectionBuildAndTest.groovy

```diff
@@ -129,11 +129,11 @@ def call(params) {
         }
       }
 
-      if (config.pactProviderVerificationsEnabled && isOnMaster) {
-        pcr.callAround('pact-provider-verification') {
-          builder.runProviderVerification(pactBrokerUrl, version)
-        }
-      }
+      /*
```

Comment:
So the problem is this code was put together with this idea:
step 1 -> create consumer pacts, 
step 2 -> verify them against a provider for **this** service

but what is actually happening is, 
step 1 -> create consumer pacts if this is a consumer
step 2 -> run provider verification if this is a provider

There is no reason for an ordered function call here, and there is no reason for this to happen at this point.
This only makes sense if the pipeline can trigger a pact verification on the provider for the service and then callback with the response and block the build from continuing on a failure.

I have removed the pact Provider Verification from here and put it after the functional tests so that a service can then run the pact provider verification against its new instance, not a pre-existing one

Reply:
remove the comment?

Reply:
right, cool ill get rid of it